src/gluonts/itertools.py

```diff
@@ -452,3 +452,31 @@ def power_set(iterable):
     return itertools.chain.from_iterable(
         itertools.combinations(iterable, r) for r in range(len(iterable) + 1)
     )
+
+
+def zip_items(*dicts, default=None, strict=False):
+    """
+    Iterate over multiple dictionary items.
+
+    >>> a = {"x": "a"}
+    >>> b = {"x": "b"}
+    >>> c = {}
+    >>> for name, (xa, xb, xc) in zip_items(a, b, c):
+    ...     print(name, xa, xb, xc)
+    x a b None
```

Comment:
This contrasts with what happens for lists with zip, in which case the output is limited to the common keys. So in this example the output iterator would be empty.

Maybe adding None could be optional?

Reply:
I guess there are generally three ways to handle this:

* be strict and require all dicts to have the same keys
* just iterate over the intersection of all keys
* iterate over the union of all keys, fall back on a default when a dict doesn't have a given key

Would we want to support all three modes?